src/auditor/filesystems.jl

```diff
@@ -43,3 +43,22 @@ function check_absolute_paths(prefix::Prefix, all_files::Vector; silent::Bool =
 
     return true
 end
+
+function ensure_executability(oh::ObjectHandle; verbose::Bool=false, silent::Bool=false)
+    if filemode(path(oh)) & 0o755 != 0o755
```

Comment:
Instead of checking for exactly `0o755`, I think we might instead want to check `filemode(path(oh)) & 0o111 == 0`, and then we probably want to set `+x` for any groups that have `+r`; so we want to set `new_mode = old_mode | (old_mode & 0o444 >> 2)`.

Reply:
I knew I was doing it wrong, I was too tired yesterday night :grimacing: Thanks!

Reply:
I'm not sure I understand why `old_mode | (old_mode & 0o444 >> 2)`:
```julia
julia> old_mode = 0o644
0x01a4

julia> string(old_mode | (old_mode & 0o444 >> 2) ; base=8)
"644"
```
doesn't look right.  I think it's just
```julia
julia> string(old_mode | 0o111; base=8)
"755"
```
no?

Reply:
you need more parens:

```
julia> old_mode = 0o640
0x01a0

julia> string(old_mode | ((old_mode & 0o444) >> 2); base=8)
"750"
```

Reply:
My bad.  :P

Reply:
`750` still doesn't have exec bit for others... or am I missing something?  Do you want to require readability in order to add the exec permission?  `chmod +x` just adds `x` everywhere

Reply:
As weird as it is, it doesn't look like a file need to be readable, in order to execute it:
```
% cat foo.c 
#include <stdio.h>

int main()
{
    printf("Hello World!\n");
    return 0;
}
% gcc -o foo foo.c
% chmod 100 foo   
% ./foo 
Hello World!
```

Reply:
Yeah, that is weird.

That being said, what I want is for us to catch obvious mistakes (e.g. `mode & 0o111 == 0`) and to fix them without accidentally giving permissions to users that it wasn't intended for.  For instance, if we have SETUID binaries in an `/sbin` folder that accidentally didn't have permissions applied (very niche scenario, I know) we wouldn't want to turn `0o600` into `0o711`; we'd want to turn it into `0o700`.

Reply:
All right, I applied the change suggested!